### PR TITLE
fix: align, taskCreate, contentTransform command

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Add a new revisionSearcher service used in Align, TaskCreate and Content transform command.
It will scroll and yield a Revisions object, on this object we can start a cool transaction.

# Before
```bash
php bin/console ems:env:align next prepub --force

Align environments
==================

Check inputs
------------

 ! [NOTE] Continuing with the source environment "next"

 ! [NOTE] Continuing with the target environment "prepub"

Execute
-------

 ! [NOTE] The source environment contains 1453 elements, start aligning environments...

 1453/1453 [============================] 100%

 ! [NOTE] 1453 revisions were already aligned


 [OK] Environments next -> prepub were aligned.



 * Duration: 108 s
 * Memory: 76 MB
````

# After

```bash
php bin/console ems:env:align next prepub --force

EMS - Environment - Align
=========================

Execute
-------

 ! [NOTE] The source environment contains 1453 elements, start aligning environments...

 1453/1453 [============================] 100%

 ! [NOTE] 1453 revisions were already aligned


 [OK] Environments next -> prepub were aligned.



 * Duration: 10 s
 * Memory: 84 MB
````

### CODE REVIEW

File: src/Command/Environment/AlignCommand.php [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    private string $searchQuery;
+
+    public const ARGUMENT_SOURCE = 'source';
+    public const ARGUMENT_TARGET = 'target';
+    public const ARGUMENT_SCROLL_SIZE = 'scrollSize';
+    public const ARGUMENT_SCROLL_TIMEOUT = 'scrollTimeout';
+    public const OPTION_FORCE = 'force';
+    public const OPTION_SEARCH_QUERY = 'searchQuery';
+    public const OPTION_SNAPSHOT = 'snapshot';
+    public const DEFAULT_SCROLL_SIZE = 100;
```
Comment (theus77) [inline src/Command/Environment/AlignCommand.php:44]: You should use the default bulk size (env variables). See other commands

File: src/Command/Environment/AlignCommand.php [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    public const DEFAULT_SCROLL_TIMEOUT = '1m';
+    public const DEFAULT_SEARCH_QUERY = '{}';
+
+    protected static $defaultName = 'ems:environment:align';
```
Comment (theus77) [inline src/Command/Environment/AlignCommand.php:48]:
```suggestion
    protected static $defaultName = 'emsco:environment:align';
```

File: src/Command/Environment/AlignCommand.php [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    private function createSearch(): Search
+    {
+        $search = $this->elasticaService->convertElasticsearchBody(
+            [$this->source->getAlias()],
+            [],
+            ['query' => Json::decode($this->searchQuery)]
+        );
+        $search->setSize($this->scrollSize);
```
Comment (theus77) [inline src/Command/Environment/AlignCommand.php:186]: Do you specify any _source fieldds? I think thta this ;etho should only get _id and _contenttype fields

Comment (Davidmattei) [reply src/Command/Environment/AlignCommand.php:186]: 👍

File: src/Command/Environment/AlignCommand.php (hunk 1 above)
Comment (Davidmattei) [reply src/Command/Environment/AlignCommand.php]: Add a revisionSearcher service, he is now aware of the default scroll size

File: src/Command/Environment/AlignCommand.php (hunk 2 above)
Comment (Davidmattei) [reply src/Command/Environment/AlignCommand.php]: Fixed

File: src/Command/Environment/AlignCommand.php (hunk 3 above)
Comment (Davidmattei) [reply src/Command/Environment/AlignCommand.php]: Good remark, indeed we only need _id and _contentType

**Review by coppee — DISMISSED** [review]

**Pull request merged**